app/controllers/application_controller.rb

```diff
@@ -5,6 +5,14 @@ class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_filter :configure_permitted_parameters, if: :devise_controller?
 
+  # SSL Production Config
+  if Rails.env.production?
+    # Force SSL except in excepted routes
+    force_ssl unless: :ssl_excepted?
+    # Reset session when hitting excepted routes so as not to leak cookie
+    before_action :reset_session, if: :ssl_excepted?
+  end
+
```

Comment:
I think I would knock up a couple of controller specs to check that it's hitting the force_ssl when expected, and not hitting it when not expected.  You can mock the response to Rails.env.production? to trick it into thinking its in production mode.